packages/playground/blueprints/src/lib/compile.ts

```diff
@@ -48,15 +48,36 @@ export interface CompiledBlueprint {
 
 export type OnStepCompleted = (output: any, step: StepDefinition) => any;
 
+export type CompletedStatus = Map<StepDefinition, boolean>;
+export type OnStepsCompleted = (completedStatus: CompletedStatus) => void;
+export type OnRunningSteps = (progressMap: PromiseMap) => void;
+
+export type PromiseMap = Map<StepDefinition, Promise<void>>;
+export type PromiseExecutor = {
+	resolve: (result: any) => void;
+	reject: (error: any) => void;
+};
```

Comment:
I don't particularly find these custom types that much more helpful than the concrete types they alias. e.g.

```ts
export type OnStepsCompleted = (completedStatus: Map<StepDefinition, boolean>) => void;
export type OnRunningSteps = (progressMap: Map<StepDefinition, Promise<void>) => void;
```

this is succinct and I don't have to make sure the named types are hiding anything or doing anything surprising.

I guess you found otherwise? what benefit do you see from having them named? you think it's a worthwhile tradeoff?

Reply:
as with the naming, I think there's room for harmony in the name of the argument. I like `completedStatus` because it indicates what it means, but even though I originally wrote `progressMap` I find it a bit low-level and internal. maybe `progressStatus`?

Reply:
I consolidated the changes into a single `onStepProgress` callback, similar in concept to `onStepCompleted`.